consensus/ethash/consensus.go

```diff
@@ -180,14 +180,18 @@ func (ethash *Ethash) VerifyHeaders(chain consensus.ChainHeaderReader, headers [
 
 func (ethash *Ethash) verifyHeaderWorker(chain consensus.ChainHeaderReader, headers []*types.Header, seals []bool, index int, unixNow int64) error {
 	var parent *types.Header
-	if index == 0 {
+	if index == 0 { // figure out how to handle genesis case
 		parent = chain.GetHeader(headers[0].ParentHash[types.QuaiNetworkContext], headers[0].Number[types.QuaiNetworkContext].Uint64()-1)
 	} else if headers[index-1].Hash() == headers[index].ParentHash[types.QuaiNetworkContext] {
 		parent = headers[index-1]
 	}
 	if parent == nil {
-		return consensus.ErrUnknownAncestor
+		if index != 0 {
+			fmt.Println("for", index, "wanted", headers[index].ParentHash[types.QuaiNetworkContext], "got", headers[index-1].Hash())
+			return consensus.ErrUnknownAncestor
+		}
```

Comment:
We don't need this exception anymore, do we? Ideally this PR shouldn't modify `consensus.go`

Reply:
Reverted to main.